llvm/tools/llvm-foreach/llvm-foreach.cpp

```diff
@@ -79,6 +92,31 @@ static void error(std::error_code EC, const Twine &Prefix) {
     error(Prefix + ": " + EC.message());
 }
 
+// With BlockingWait=false this function just goes through the all
+// submitted jobs to check if some of them have finished.
+int checkIfJobsAreFinished(std::list<sys::ProcessInfo> &JobsSubmitted,
+                           bool BlockingWait = true) {
+  std::string ErrMsg;
+  auto It = JobsSubmitted.begin();
```

Comment:
Is it possible to not use `auto` here?

Reply:
It's possible, but I'm not sure why it's needed here. What's the problem of `auto`?

Reply:
It is just LLVM coding standards suggesting not use auto when the type is not obvious. https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable

Reply:
Okay, I thought it's obvious from the function signature which is just above. I'd like not to change it to save readability